truss/templates/trtllm-audio/packages/whisper_trt/batching.py

```diff
@@ -0,0 +1,48 @@
+from typing import TYPE_CHECKING
+import logging
+
+if TYPE_CHECKING:
+    from whisper_trt import WhisperModel
+
+from whisper_trt.types import BatchWhisperItem, DEFAULT_NUM_BEAMS
+
+from async_batcher.batcher import AsyncBatcher
+from torch import Tensor
+import torch
+
+FIXED_TEXT_PRFIX = "<|startoftranscript|><|en|><|transcribe|><|0.00|>"
+
+
+class WhisperBatchProcessor(AsyncBatcher[list[BatchWhisperItem], list[str]]):
+    def __init__(self, model, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model: "WhisperModel" = model
+
+    def concat_and_pad_mels(self, tensors: list[Tensor]):
+        while len(tensors) < self.max_batch_size:
+            tensors.append(tensors[-1])
+        res = torch.cat(tensors, dim=0).type(torch.float16)
+        return res
+
+    def concat_and_pad_prompts(self, prompts: list[list]) -> Tensor:
```

Comment:
Doc comment indicating that modifications are done in place and using the past prompt_id.